lib/lambda_ethereum_consensus/beacon/pending_blocks.ex

```diff
@@ -114,6 +114,12 @@ defmodule LambdaEthereumConsensus.Beacon.PendingBlocks do
     end)
   end
 
+  @impl true
```

Comment:
little bug, we were calling `BlockDownloader.request_blocks_by_root()` with an empty array sometimes

Reply:
This wouldn't work:
```elixir
iex(1)> %MapSet{} = MapSet.new([1])
MapSet.new([1])
iex(2)> %MapSet{} = MapSet.new([])
MapSet.new([])
```

[This SO question might have ways to do it](https://stackoverflow.com/questions/49284613/pattern-match-an-empty-mapset)

Reply:
wow, good catch! thanks

Reply:
Got bit by it more than one time 😞